openupgrade_scripts/scripts/base/17.0.1.3/pre-migration.py

```diff
@@ -123,9 +123,13 @@ def _update_partner_private_type(cr):
     openupgrade.logged_query(
         cr,
         """
-        UPDATE res_partner
-        SET type = 'contact'
-        WHERE type = 'private'
+            UPDATE res_partner
+            SET type = CASE
+                WHEN type = 'private' AND name IS NOT NULL THEN 'contact'
+                WHEN type = 'private' AND name IS NULL THEN 'other'
```

Comment:
phía dưới có WHERE rồi
```suggestion
                WHEN name IS NOT NULL THEN 'contact'
                WHEN name IS NULL THEN 'other'
```